### PR TITLE
add initial contributing guidelines

### LINKED ISSUE #307: Write up plan for ongoing work
**What needs doing?**

We need to write up a plan for how we work on the Neontribe website outside of agreed up sprints. Including how Github is managed.

**Plan**
- [ ] Agree on how we plan and manage the website outside of sprints
- [ ] Write up this agreement

**Acceptance Criteria*
- [ ] Documentation exists on how Neontribers can work on the website

### PR DESCRIPTION
**Pull request by RpprRoger**
Fixes #307

WIP guidelines which can be used as a talking point

## Description

- Adds initial stab at contrib guidelines

Feedback welcome!

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+1. Assign yourself and move the ticket to the In progress column.
+1. Create a branch with a name to help match it to the ticket, e.g.: `307-contrib-docs`. _n.b._ [avoid slashes in branch names](https://stackoverflow.com/questions/2527355/using-the-slash-character-in-git-branch-name/2527452#2527452)
+1. Work on the feature branch, make small commits which include the _reason_ for each commit's change.
+1. Create a pull request when you're happy with the changeset, and happy that the ticket passes the ticket's [Definition of done](#definition-of-done).
+   _n.b._ Pull requests are the pull requesters responsibility, when it recieves the request number
+1. Await code review, see [Review proces](#review-process)
+
+Please make sure that you keep you activity up-to-date:
+
+- only assign one ticket at a time to yourself
```
Comment (katjam) [inline CONTRIBUTING.md]: Maybe we can incorporate these 3 into the when contributing list? e.g. the branch name is already there.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+When contributing to this project please do the following:
+
+1. Pick something near the top of the to do column, which you feel you have the time and confidence to tackle.
+   - We expect you to learn while tackling issues, please don't be put off by a tasks' seeming complexity.
+   - If you need more information please follow up with the person who spawned the task.
+   - Check the [Definition of done](#definition-of-done) for the ticket
+1. Assign yourself and move the ticket to the In progress column.
+1. Create a branch with a name to help match it to the ticket, e.g.: `307-contrib-docs`. _n.b._ [avoid slashes in branch names](https://stackoverflow.com/questions/2527355/using-the-slash-character-in-git-branch-name/2527452#2527452)
+1. Work on the feature branch, make small commits which include the _reason_ for each commit's change.
```
Comment (katjam) [inline CONTRIBUTING.md]: Maybe a tip - don't forget to include tests if appropriate

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+Please make sure that you keep you activity up-to-date:
+
+- only assign one ticket at a time to yourself
+- name a branch based upon the ticket
+- if the ticket needs to be put on hold
+
+---
+
+WIP:
```
Comment (katjam) [inline CONTRIBUTING.md]: Could tidy the WIP section grammatically with uppercase letters and lists even where only one item - for better readability/ consistency

File: CONTRIBUTING.md (hunk 2 above)
Comment (katjam) [inline CONTRIBUTING.md]: Additionally, I like these: https://chris.beams.io/posts/git-commit/

Probably not necessary to include that specific link but the most important one for me is:
https://chris.beams.io/posts/git-commit/#imperative

**Comment by Nikomus** [comment]
@RpprRoger We talked about this briefly at the last weekly meeting and some things that could be added/talked about more would be:
- Use of `good first issue` labels - to help newer folk choose appropriate tickets
- Doing `EEE/VVV` ratings - for the same
- Resolving use of project board and ordered milestone - whether we should have both or not, how to promote and relegate between the two

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+1. Pick something near the top of the to do column, which you feel you have the time and confidence to tackle.
+   - We expect you to learn while tackling issues, please don't be put off by a tasks' seeming complexity.
+   - If you need more information please follow up with the person who spawned the task.
+   - Check the [Definition of done](#definition-of-done) for the ticket
+1. Assign yourself and move the ticket to the In progress column.
+1. Create a branch with a name to help match it to the ticket, e.g.: `307-contrib-docs`. _n.b._ [avoid slashes in branch names](https://stackoverflow.com/questions/2527355/using-the-slash-character-in-git-branch-name/2527452#2527452)
+1. Work on the feature branch, make small commits which include the _reason_ for each commit's change.
+1. Create a pull request when you're happy with the changeset, and happy that the ticket passes the ticket's [Definition of done](#definition-of-done).
+   _n.b._ Pull requests are the pull requesters responsibility, when it recieves the request number
```
Comment (Nikomus) [inline CONTRIBUTING.md]:
```suggestion
   _n.b._ Pull requests are the pull requester's responsibility, when it receives the request number
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+## Getting tickets ready to be worked on
+
+- weekly meeting to discuss open tickets
+- tickets should be left in the backlog or closed if vetoed
+- anyone can create issues
+- creating an issue involves writing up a compelling argument for a feature, perhaps pictures. Adding a definition of done is important, doesn't have to be acceptance criteria
+
+## Review process
+
+2 or more peers need to approve the commit, when reviewing check the code for errors and check that it achieves the definition of done for the ticket
```
Comment (Nikomus) [inline CONTRIBUTING.md]:
```suggestion
Two or more peers need to approve the commit, when reviewing check the code for errors and check that it achieves the definition of done for the ticket
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+## Testing
+
+is done using feature branch deploys on pull request
+
+## Definition of done
+
+Acceptance critera etc.
```
Comment (Nikomus) [inline CONTRIBUTING.md]:
```suggestion
Acceptance criteria etc.
```

**Comment by RpprRoger** [comment]
@AlinChindea Reminded me of the README, link to it from this CONTRIBUTING doc

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+Ongoing development on this project is managed by this project board: https://github.com/neontribe/www/projects/6.
+
+## Getting started on development
+
+Any issues in the `To do` column are assumed to be important enough and have been prioritised.
+
+When contributing to this project please do the following:
+
+1. Pick something near the top of the to do column, which you feel you have the time and confidence to tackle.
```
Comment (AlinChindea) [inline CONTRIBUTING.md]:
```suggestion
1. Pick something near the top of the `To do` column, which you feel you have the time and confidence to tackle.
```

Consistency :-)

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+   - We expect you to learn while tackling issues, please don't be put off by a tasks' seeming complexity.
+   - If you need more information please follow up with the person who spawned the task.
+   - Check the [Definition of done](#definition-of-done) for the ticket
+1. Assign yourself and move the ticket to the In progress column.
```
Comment (AlinChindea) [inline CONTRIBUTING.md]:
```suggestion
1. Assign yourself and move the ticket to the `In progress` column.
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+1. Create a branch with a name to help match it to the ticket, e.g.: `307-contrib-docs`. _n.b._ [avoid slashes in branch names](https://stackoverflow.com/questions/2527355/using-the-slash-character-in-git-branch-name/2527452#2527452)
+1. Work on the feature branch, make small commits which include the _reason_ for each commit's change.
+1. Create a pull request when you're happy with the changeset, and happy that the ticket passes the ticket's [Definition of done](#definition-of-done).
+   _n.b._ Pull requests are the pull requesters responsibility, when it recieves the request number
+1. Await code review, see [Review proces](#review-process)
+
+Please make sure that you keep you activity up-to-date:
```
Comment (AlinChindea) [inline CONTRIBUTING.md]:
```suggestion
Please make sure that you keep your activity up-to-date:
```

Or maybe that's not needed at all? As in, you've added the word 'you' twice?

File: CONTRIBUTING.md (hunk 5 above)
Comment (AlinChindea) [reply CONTRIBUTING.md]: To me, these are in fact 3 separate points that can be split over three lines (would improve readability, too):
```
- Two or more peers need to approve the commit 
- When reviewing, check the code for errors 
- Check that it achieves the definition of done for the ticket
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+- only assign one ticket at a time to yourself
+- name a branch based upon the ticket
+- if the ticket needs to be put on hold
```
Comment (AlinChindea) [inline CONTRIBUTING.md]: seems like an unfinished point - I see an `if` I expect a then or an else :smiley:

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+## Getting started on development
+
+Any issues in the `To do` column are assumed to be important enough and have been prioritised.
+
+When contributing to this project please do the following:
+
+1. Pick something near the top of the to do column, which you feel you have the time and confidence to tackle.
+   - We expect you to learn while tackling issues, please don't be put off by a tasks' seeming complexity.
+   - If you need more information please follow up with the person who spawned the task.
```
Comment (katquatermass) [inline CONTRIBUTING.md]:
```suggestion
   - If you need more information please follow up with the person who spawned/created the task.
```
I almost went for a change to plain English and got rid of spawned - but then thought actually, no its good for people to learn coders language as they start working on this - what do people feel about using both the coders slang term and the plain english term for things like this?

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+---
+
+WIP:
+
+## Getting tickets ready to be worked on
+
+- weekly meeting to discuss open tickets
+- tickets should be left in the backlog or closed if vetoed
+- anyone can create issues
+- creating an issue involves writing up a compelling argument for a feature, perhaps pictures. Adding a definition of done is important, doesn't have to be acceptance criteria
```
Comment (katquatermass) [inline CONTRIBUTING.md]:
```suggestion
- creating an issue involves writing up a compelling argument for an improvement, perhaps including pictures if it is a change to the existing site. Anything new should be supported by a user story that will help to inform a definition of done. This definition of done is important, but does not have to be full acceptance criteria at this stage.
```
It had the word feature in. A little part of me cried. Have I taken it too far the other way?

File: CONTRIBUTING.md (hunk 10 above)
Comment (katquatermass) [reply CONTRIBUTING.md]: Is the second half an instruction for how to label if it has been put on hold?

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,49 @@
+# Contribution guidelines
+
+Ongoing development on this project is managed by this project board: https://github.com/neontribe/www/projects/6.
+
+## Getting started on development
+
+Any issues in the `To do` column are assumed to be important enough and have been prioritised.
```
Comment (katquatermass) [inline CONTRIBUTING.md]: Is going to need changing if today's meeting approves the removal of the "to do" column

**Comment by katjam** [comment]
Closing as @RpprRoger and @katquatermass are still working on this.

**Pull request closed without merging**